### PR TITLE
Creat models for my database

### PR DESCRIPTION
**Pull request by ruxmur**

### CODE REVIEW

File: app/Models/about_client.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class about_client extends Model
```
Comment (vistrati) [inline app/Models/about_client.php:8]: please name in in camelCase `AboutCLient`

File: app/Models/client.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        'start_time',
+        'end_time'
+    ];
+
+    public function membership()
+    {
+        return $this->hasMany('membership');
+    }
+
+    public function about_client()
```
Comment (vistrati) [inline app/Models/client.php:26]: `aboutCLlient`

**Review by vistrati — CHANGES_REQUESTED** [review]

File: app/Models/client.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    protected $fillable = [
+        'id',
+        'first_name',
+        'last_name',
+        'membership_id',
+        'start_time',
+        'end_time'
+    ];
+
+    public function membership()
```
Comment (vistrati) [inline app/Models/client.php:21]: `memberships`, if has many

File: app/Models/client.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        return $this->hasMany('membership');
+    }
+
+    public function aboutClient()
+    {
+        return $this->hasOne('aboutClient');
+    }
+
+    public function reservation()
```
Comment (vistrati) [inline app/Models/client.php:31]: `reservations`

**Review by vistrati — CHANGES_REQUESTED** [review]

File: app/Models/reservation.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        'client_id',
+        'workout_types_id',
+        'coaches_id',
+        'start_time',
+        'end_time'
+    ];
+
+    public function clients()
+    {
+        return $this->belongsToMany('client');
```
[the remark this answers is not in the document]
Comment (vistrati) [reply app/Models/reservation.php:23]: PLease pay more attention to relations
https://laravel.com/docs/8.x/eloquent-relationships#one-to-many-inverse

File: app/Models/reservation.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public function workout_types()
+    {
+        return $this->belongsToMany('workoutType');
```
[the remark this answers is not in the document]
Comment (vistrati) [reply app/Models/reservation.php:28]: Please pay more attention to relation types
https://laravel.com/docs/8.x/eloquent-relationships#one-to-many-inverse

File: app/Models/workout_type.php [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    use HasFactory;
+
+    protected $fillable = [
+        'id',
+        'name'
+    ];
+
+    public function reservation()
+    {
+        return $this->belongsTo('reservation');
```
[the remark this answers is not in the document]
Comment (vistrati) [reply app/Models/workout_type.php:19]: please pay more attention to relation.
https://laravel.com/docs/8.x/eloquent-relationships#one-to-many-inverse

File: app/Models/workout_type.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public function coachs()
+    {
+        return $this->belongsToMany('coach');
```
[the remark this answers is not in the document]
Comment (vistrati) [reply app/Models/workout_type.php:24]: If you use different table names for pivot table then you should specify the name
https://laravel.com/docs/8.x/eloquent-relationships#many-to-many-model-structure

Screenshot (53)

File: app/Models/workout_type.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public function clients()
+    {
+        return $this->belongsToMany('client');
```
[the remark this answers is not in the document]
Comment (vistrati) [reply app/Models/workout_type.php:29]: If you use different table names for pivot table then you should specify the name
https://laravel.com/docs/8.x/eloquent-relationships#many-to-many-model-structure

Screenshot (53)

**Review by vistrati — CHANGES_REQUESTED** [review]

File: app/Models/reservation.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        'start_time',
+        'end_time'
+    ];
+
+    public function clients()
+    {
+        return $this->belongsToMany('client');
+    }
+
+    public function workout_types()
```
[the remark this answers is not in the document]
Comment (ruxmur) [reply app/Models/reservation.php]: all mistakes have been corrected. thank you

**Pull request merged**